src/amo/reducers/landing.js

```diff
@@ -0,0 +1,36 @@
+import {
+  LANDING_GET,
+  LANDING_LOADED,
+  LANDING_FAILED,
+} from 'core/constants';
+
+
+export const initialState = {
+  featured: { count: 0, results: [] },
+  highlyRated: { count: 0, results: [] },
+  loading: false,
+  popular: { count: 0, results: [] },
+};
+
+export default function landing(state = initialState, action) {
+  const { payload } = action;
+  let newState;
+  switch (action.type) {
+    case LANDING_GET:
+      return { ...state, ...payload, loading: true };
```

Comment:
I would rather see `addonType: payload.addonType` instead of `...payload` because it's not super clear otherwise. Should `initialState` have a default value for `addonType`? Maybe that would be more clear.